sphinx/ext/autosummary/__init__.py

```diff
@@ -770,7 +770,7 @@ def run(self) -> tuple[list[Node], list[system_message]]:
 
 def get_rst_suffix(app: Sphinx) -> str | None:
     def get_supported_format(suffix: str) -> tuple[str, ...]:
-        parser_class = app.registry.get_source_parsers().get(suffix)
+        parser_class = app.registry.get_source_parsers().get(suffix.lstrip('.'))
```

Comment:
FYI: `suffix.removeprefix('.')` is better than this but it requires Python 3.9 or later. Python 3.8 haven't reached EOL yet.

Reply:
We don't support 3.8 so you can go for this (and thank you for reminding me that this one exists!)

Reply:
Now, I'm wondering:

- With `removeprefix`, you only remove a single occurrence of the dot, so if the user miswrites `..md` for instance, you'll get `.md`.
- With `.lstrip`, you'll end up with `md` anyway, which (AFAICT) is what the user wants.

So perhaps we can keep the `lstrip` actually

Reply:
> We don't support 3.8

Oh, sorry. I missed it.

Regarding to `..md` input, I think that it's an invalid `source_suffix` configuration. So I think that we don't need to support the case. But OK. I'll keep `lstrip` here.

Reply:
Actually, I'm just wondering whether we should silently suppress the error (with `lstrip`) or have an explicit error if there is still `.` after doing `removeprefix`.

Reply:
@chrisjsewell What do you think about this case?

Reply:
I'm relatively abivelent to be honest, I guess `suffix.removeprefix('.')` is technically the best 👍 
I don't feel we need to add a warning here, that would be more something to do in a "top-level" configuration validator

Reply:
OK. I've changed to use `removeprefix`.

Is this ready to merge now?

Reply:
looks good thanks, just waiting to fix the flaky test failure before merging